packages/eslint-plugin/tests/rules/thenable-in-promise-aggregators.test.ts

```diff
@@ -0,0 +1,358 @@
+import { RuleTester } from '@typescript-eslint/rule-tester';
+
+import rule from '../../src/rules/thenable-in-promise-aggregators';
+import { getFixturesRootDir } from '../RuleTester';
+
+const rootDir = getFixturesRootDir();
+const messageIdInArray = 'inArray';
+const messageIdArrayArg = 'arrayArg';
+const messageIdNonArrayArg = 'nonArrayArg';
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2018,
+    tsconfigRootDir: rootDir,
+    project: './tsconfig.json',
+  },
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('thenable-in-promise-aggregators', rule, {
+  valid: [
+    `
+async function test() {
+  await Promise.race([Promise.resolve(3)]);
+}
+    `,
+    `
+async function test() {
+  await Promise.all([Promise.resolve(3)]);
+}
+    `,
+    `
+async function test() {
+  await Promise.allSettled([Promise.resolve(3)]);
+}
+    `,
+    `
+async function test() {
+  await Promise.race([(async () => true)()]);
+}
+    `,
+    `
+async function test() {
+  function returnsPromise() {
+    return Promise.resolve('value');
+  }
+  await Promise.race([returnsPromise()]);
+}
+    `,
+    `
+async function test() {
+  async function returnsPromiseAsync() {}
+  await Promise.race([returnsPromiseAsync()]);
+}
+    `,
+    `
+async function test() {
+  let anyValue: any;
+  await Promise.race([anyValue]);
+}
+    `,
+    `
+async function test() {
+  let unknownValue: unknown;
+  await Promise.race([unknownValue]);
+}
+    `,
+    `
+async function test() {
+  const numberPromise: Promise<number>;
+  await Promise.race([numberPromise]);
+}
+    `,
+    `
+async function test() {
+  class Foo extends Promise<number> {}
+  const foo: Foo = Foo.resolve(2);
+  await Promise.race([foo]);
+
+  class Bar extends Foo {}
+  const bar: Bar = Bar.resolve(2);
+  await Promise.race([bar]);
+}
+    `,
```

Comment:
Ref: https://typescript-eslint.io/contributing/local-development#granular-unit-tests

We prefer test cases to be small and self-contained, rather than having multiple assertions in one case.
Could you please split this case?

Reply:
Ahh! Copied these from `await-thenable` almost precisely, but happy to split